website/client/src/components/shops/buyModal.vue

```diff
@@ -511,6 +545,18 @@ export default {
     nonSubscriberHourglasses () {
       return (!this.user.purchased.plan.customerId && !this.user.purchased.plan.consecutive.trinkets && this.getPriceClass() === 'hourglasses');
     },
+    itemAvailable () {
+      // for card types the item has the yearRound flag
+      if (this.item.purchaseType === 'card') {
+        return this.item.yearRound;
+      }
+      // Check if it has the canBuy property of type boolean. If not then
+      // we just return true.
+      if (this.item && (typeof this.item.canBuy === 'boolean')) {
```

Comment:
why is this checking for `this.item` since it's already used in the check above so if it was undefined line 550 would have already thrown an error